### ISSUE TITLE
[Data source] Wikidata localization

### ISSUE THREAD

**Issue by yarons** [WeblateOrg/weblate#10514]
### Describe the problem

Sometimes I need to translate large sets of data from Wikidata, doing it manually is very time consuming.

### Describe the solution you'd like

Adding an option for a list of objects or a SPARQL query to fetch all the relevant data for translation, the translated terms will (or should) be directly saved back to Wikidata (another option is to create an internal repository and push on request which may lead to collisions on larger scales).

### Describe alternatives you've considered

Translating directly in Wikidata, it's possible but very time-consuming.

### Additional context

I actually have a task waiting just for that, I'm correlating the data from OpenStreetMap to Wikidata, so now whenever I'm changing the data in Wikidata it's being automatically reflected in OSM.
This way I can have lists of countries/regions/cities/streets and translate them using Zen mode instead of going one by one, so the only difficult part here is to correlate the names of the city to the Wikidata properties (which should be done once) and then translate.
We can also have supporting languages so we can fetch the main name in addition to the English translation/transliteration to help with proper translation.

It will also help with translating all the terms in the OSM wiki (which is sort of a Wikibase with different data correlations).

Thanks :)

**Comment by nijel** [comment]
Yes, technically it would be possible (we do something similar in the CDN add-on), but I'm not really convinced it is a good idea to embed this into Weblate.

**Comment by yarons** [comment]
Cool, I know that translatewiki are working on something: https://phabricator.wikimedia.org/T64695
